src/Sign.SignatureProviders.KeyVault/KeyVaultService.cs

```diff
@@ -48,50 +51,61 @@ public void Dispose()
 
         public async Task<X509Certificate2> GetCertificateAsync(CancellationToken cancellationToken)
         {
-            KeyVaultCertificateWithPolicy certificateWithPolicy = await GetCertificateWithPolicyAsync(cancellationToken);
+            KeyVaultCertificate certificateWithPolicy = await GetCertificateVersionAsync(cancellationToken);
 
             return new X509Certificate2(certificateWithPolicy.Cer);
         }
 
         public async Task<RSA> GetRsaAsync(CancellationToken cancellationToken)
         {
-            KeyVaultCertificateWithPolicy certificateWithPolicy = await GetCertificateWithPolicyAsync(cancellationToken);
+            KeyVaultCertificate certificateWithPolicy = await GetCertificateVersionAsync(cancellationToken);
 
             CryptographyClient cryptoClient = new(certificateWithPolicy.KeyId, _tokenCredential);
             return await cryptoClient.CreateRSAAsync(cancellationToken);
         }
 
-        private async Task<KeyVaultCertificateWithPolicy> GetCertificateWithPolicyAsync(CancellationToken cancellationToken)
+        private async Task<KeyVaultCertificate> GetCertificateVersionAsync(CancellationToken cancellationToken)
         {
-            if (_certificateWithPolicy is not null)
+            if (_certificate is not null)
             {
-                return _certificateWithPolicy;
+                return _certificate;
             }
 
             await _mutex.WaitAsync(cancellationToken);
 
             try
             {
-                if (_certificateWithPolicy is null)
+                if (_certificate is null)
                 {
                     Stopwatch stopwatch = Stopwatch.StartNew();
 
                     _logger.LogTrace(Resources.FetchingCertificate);
 
                     CertificateClient client = new(_keyVaultUrl, _tokenCredential);
-                    Response<KeyVaultCertificateWithPolicy> response = await client.GetCertificateAsync(_certificateName, cancellationToken);
+                    if(_certificateVersion is null)
```

Comment:
Because empty strings also seem to be allowed you should probably use `if (string.IsNullOrEmpty` here instead.

Reply:
Changed to IsNullOrEmpty check